### DISCUSSION TITLE
Refresh Token cache.json

### DISCUSSION THREAD

**Discussion by Lockwi** [tdorssers/TeslaPy#47] [Q&A]
One question: I know a few weeks ago the programme created a cache.json. The refresh token was found in it. Today I installed the new version, but I can no longer find the json file. Is there a way to display the token? Many thanks for feedback.

**Comment by tdorssers** [comment]
Try:
```python
from teslapy import Tesla

with Tesla('elon@tesla.com') as tesla:
    print(tesla.token)
```

**Reply by Lockwi** [reply]
hm I don´t know how. I know the refresh_token starts with qts- or eu-
I got the message "none" if I use the print

I see in the debug mode in gui.py some token. Access, id refresh, but not the short one only the large ones.

gui.py and cli.py are working fine. But I don´t see anymore the refresh_token. the short one.

**Reply by tdorssers** [reply]
Ah, that's not the refresh token, that's the JWT token and TeslaPy doesn't use that token anymore. Only the 'SSO' token is used, which includes a refresh_token too.
If you get `none`, please add `tesla.fetch_token()` before print.
